airflow/utils/db.py

```diff
@@ -1194,7 +1194,7 @@ def check_run_id_null(session: Session) -> Iterable[str]:
     invalid_dagrun_filter = or_(
         dagrun_table.c.dag_id.is_(None),
         dagrun_table.c.run_id.is_(None),
-        dagrun_table.c.execution_date.is_(None),
+        dagrun_table.c.logical_date.is_(None),
```

Comment:
This is for data integrity checks in `upgradedb`. But I’m thinking maybe we can just drop the checks entirely?

If we _require_ users to upgrade to 2.10 (or even 2.11) before upgrading, they should have already run those checks and have data integrity (enforced by the database) when they upgrade to 3.0.

(Same for the column renames below.)

Reply:
I raised #41850 to remove the checks entirely.